### PR TITLE
feat: Add support for Snowflake connections with private keys

### LINKED ISSUE #409: Support private key-based connections to Snowflake
**Is your feature request related to a problem? Please describe.**
It is not possible to connect to a Snowflake instance where a user's authentication happens via an RSA private key, this is preventing us from using dbt-sugar.

**Describe the solution you'd like**
dbt-sugar to support RSA private keys for Snowflake connections

**Describe alternatives you've considered**
The alternative is setting up a different user login method, but this is less secure.

### PR DESCRIPTION
**Pull request by badge**
# Description

This PR adds support for Snowflake connections that use RSA private keys instead of passwords. The `SnowflakeDbtProfilesModel` class has been updated to make `password` optional, and a new `private_key` attribute, also optional, has been added. A new root validator then checks that one of these is present.

# How was the change tested

Tested locally, and added a test Snowflake profile using a private key instead of a password.

# Issue Information

Closes #409

# Checklist

(Ideally, all boxes are checked by the time we merged the PR, if you don't know how to do any of these don't hesitate to say so in the PR and we'll help you out.)

- [x] I formatted my PR name according to [CONTRIBUTING.md](CONTRIBUTING.md)
- [x] I added a news fragment to help populating the changelog as encouraged in [CONTRIBUTING.md](CONTRIBUTING.md)
- [x] I added "Closes #<issue_number>" in the "Issue Information" section (if no issue, feel free to tick thick the box anyway).

### CODE REVIEW

File: dbt_sugar/core/connectors/snowflake_connector.py [hunk 1]
```diff
@@ -30,15 +30,28 @@ def __init__(
...
-        self.connection_url = URL(
-            drivername="postgresql+psycopg2",
-            user=connection_params.get("user", str()),
-            password=connection_params.get("password", str()),
-            database=connection_params.get("database", str()),
-            account=connection_params.get("account", str()),
-            warehouse=connection_params.get("warehouse", str()),
-        )
-        self.engine = sqlalchemy.create_engine(self.connection_url)
+
```
Comment (bastienboutonnet) [inline dbt_sugar/core/connectors/snowflake_connector.py:33]: I think this causes flake8 in my pre-commit.ci run to complain could you remove that blank line?

**Review by bastienboutonnet — COMMENTED** [review]
One tiny flake 8 issue, the rest looks really really stellar to me! Thanks

File: dbt_sugar/core/connectors/snowflake_connector.py (hunk 1 above)
Comment (badge) [reply dbt_sugar/core/connectors/snowflake_connector.py]: Done! 🥳

**Review by bastienboutonnet — APPROVED** [review]
Thanks a lot for this @badge I'll merge it tonight and cut a new release with your changes. I'll update you in the issue when that is done. If you don't get any updates, feel free to holler at me, I may have forgotten

 Cat says shipit!

**Pull request merged**